kratos/utilities/amgcl_csr_conversion_utilities.h

```diff
@@ -69,15 +69,30 @@ class AmgclCSRConversionUtilities
             pAconverted->SetIsOwnerOfData(false);
         }
         else{ //if rA is the owner, transfer ownership to the csr_matrix
-            rA.own_data = false;
+            //rA.own_data = false;
             pAconverted->SetIsOwnerOfData(true);
         }
 
+        //assume ownership of data via unique_ptr (it will be assigned to the converted matrix)
+        Kratos::unique_ptr<TIndexType[]> pRowData((TIndexType*)(rA.ptr));
+        Kratos::unique_ptr<TIndexType[]> pColData((TIndexType*)(rA.col));
+        Kratos::unique_ptr<TDataType[]> pValueData((TDataType*)(rA.val));
+
         pAconverted->SetRowSize(rA.nrows);
         pAconverted->SetColSize(rA.ncols);
-        pAconverted->AssignIndex1Data((TIndexType*)(rA.ptr), rA.nrows+1);
-        pAconverted->AssignIndex2Data((TIndexType*)(rA.col), rA.nnz);
-        pAconverted->AssignValueData((TDataType*)(rA.val), rA.nnz);
+        pAconverted->AssignIndex1Data(pRowData, rA.nrows+1);
+        pAconverted->AssignIndex2Data(pColData, rA.nnz);
+        pAconverted->AssignValueData(pValueData, rA.nnz);
+
+        if(rA.own_data == false){ //if rA is not the owner we must detach the pointers
+            pRowData.release(); //we don't want it to be deleted here
```

Comment:
this is the "strange" use of unique_ptr i am obliged to do in this PR. 
The point is that amgcl doesn't have a unique_ptr so i need to create one. However if amgcl does not have ownership of it, neither amgcl nor the csr_matrix it converts to can do the deleting